backend-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/TorClient.kt

```diff
@@ -80,49 +82,62 @@ class TorClient private constructor(
      * This client is isolated from any other Tor usage, and queries made with this client
      * are isolated from each other (but may still be correlatable by the server through
      * request timing, if the caller does not mitigate timing attacks).
+     *
+     * @return given client or null if tor instantiation failed
      */
-    suspend fun createIsolatedWalletClient(endpoint: String): PartialTorWalletClient =
-        accessMutex.withLock {
-            checkNotNull(nativeHandle) { "TorClient is disposed" }
-            IsolatedTorWalletClient.new(
-                isolatedTorClient = isolatedTorClientInternal(),
-                endpoint = endpoint
-            )
-        }
+    suspend fun createIsolatedWalletClient(endpoint: String): PartialTorWalletClient? =
```

Comment:
FYI: TorClient is Optional at the root level only in iOS. `createIsolatedWalletClient` either fails or succeeds but it won't return nil in iOS. Simply said: iOS' SDK has a dependency injection system with TorClient being Optional. The instance is either registered or TorClient init failed and nil is registered instead. From that point onwards, DI.receive(TorClient.self) is the way how to get to the instance (or nil). Everything inside TorClient is no longer Optional.